test/e2e/k8petstore.go

```diff
@@ -0,0 +1,182 @@
+	}
+	//Make sure exit code != 0
+	if err = cmd.Wait(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				log.Printf("Exit Status: %d", status.ExitStatus())
+			}
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Logf("... Done starting k8petstore ")
+
+	//Lets poll until we reach min_expected transactions
+	totalTransactions := 0
+	Logf("Start polling, timeout is %v seconds", maxTime)
+
+	//How long until the FIRST transaction...
+	startupTimeout := time.After(time.Duration(k8bpsStartupTimeout))
+
+	//How long we will wait until we reach the nth (ie. 500th) transaction
+	transactionsCompleteTimeout := time.After(time.Duration(maxTime))
+	tick := time.Tick(2 * time.Second)
+	var ready = false
+
+	Logf("Now waiting %v seconds to see progress (transactions  > 3)", k8bpsStartupTimeout)
+T:
+	for {
+		select {
+		case <-transactionsCompleteTimeout:
+			//i.e. if we dont have n*1000 transactions, fail the test.
+			Logf("Timeout %v reached. Breaking!", tick)
+			break T
+		case <-startupTimeout:
+			err, totalTransactions = readTransactions(c, ns)
+			Logf("Timeout %v reached.  Checking if transactions have occured.", startupTimeout)
+			//If we don't have 3 transactions, fail the test.
+			if err != nil {
+				Logf("Failed : Error %v", err)
+				break T
+			}
+			if totalTransactions < 3 {
+				break T
+			}
+			ready = true
+		case <-tick:
+			//Pass if we've collected enough transactions
+			err, totalTransactions = readTransactions(c, ns)
+			if ready {
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if totalTransactions > minExpected {
+				break T
+			}
+			Logf("%v == %v total petstore transactions stored into redis. ==", time.Now(), totalTransactions)
+		}
+	}
+
+	//Finally! We should have exceeded the min_expected num of transactions.
+	//If this fails, but there are transactions being created, we may need to recalibrate
+	//the min_expected value - or else - your cluster is broken/slow !
+	Ω(totalTransactions).Should(BeNumerically(">", minExpected))
+}
+
+var _ = Describe("Pet Store", func() {
+
+	//Used for calibrating expected transaction #s in the soak portion.
+	var minionCount int
+	f := NewFramework("petstore")
+
+	//On a single node cluster, we expect about 10 transactions every second on average.
+	//admittedly, this is a very rough estimate given that the ETL is done in batches...
+	It(fmt.Sprintf("should scale to %v transactions in %v seconds", k8bpsSmokeTestTransactions, k8bpsSmokeTestTimeout), func() {
+		minions, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+		minionCount = len(minions.Items)
+
+		//max number of trial before k8petstore.sh dies.
+		var loadGenerators int = minionCount
+		var restServers int = minionCount
+		fmt.Printf("load generators / rest generators [ %v  /  %v ] ", loadGenerators, restServers)
+		//At least 500 transactions should be acquired within a short time period.
```

Comment:
500 transactions should be acquired rapidly.  That fact doesn't change with this parameters.  The parameters themselves should based on the fact that, the batches are sent in several hundred at a time, so as soon as the first data generator runs, we shoudl have several 100s of transactions.